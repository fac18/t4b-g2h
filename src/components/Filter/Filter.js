import React, { useState, useEffect } from "react";
import filterAll from "../../utils/filterFunctions";
import PropTypes from "prop-types";

const Filter = ({ searchResult, setFilteredResult }) => {
  // could refactor all states to useReducer
  const [collectionValue, setCollectionValue] = useState("");
  const [artistValue, setArtistValue] = useState("");
  const [periodValue, setPeriodValue] = useState("");
  const [mediumValue, setMediumValue] = useState("");
  const [orientationValue, setOrientationValue] = useState("");
  const [colourValue, setColourValue] = useState("");
  const [peopleValue, setPeopleValue] = useState("");

  const filteredAllData = filterAll(
    searchResult,
    collectionValue,
    artistValue,
    periodValue,
    mediumValue,
    orientationValue,
    peopleValue,
    colourValue
  );

  useEffect(() => {
    setFilteredResult(filteredAllData);
  }, [
    collectionValue,
    artistValue,
    periodValue,
    mediumValue,
    orientationValue,
    peopleValue,
    colourValue
  ]);

  return (
    <form>
      <label id="collectionValue">Collection</label>
      <input
        type="text"
        id="collectionValue"
        value={collectionValue}
        onChange={event => setCollectionValue(event.target.value)}
      ></input>
      <label id="artistValue">Artist/Creator</label>
      <input
        type="text"
        id="artistValue"
        value={artistValue}
        onChange={event => setArtistValue(event.target.value)}
      ></input>
      <label id="periodValue">Period/Century</label>
      <input
        type="text"
        id="periodValue"
        value={periodValue}
        onChange={event => setPeriodValue(event.target.value)}
      ></input>
      <label htmlFor="colourValue" id="colourValue">
        Colour/B&W
      </label>
      <select
        type="select"
        id="colourValue"
        value={colourValue}
        onChange={event => setColourValue(event.target.value)}
      >
        <option value="" disabled selected>
          Select an option
        </option>
        <option value="">None</option>
        <option value="Colour">Colour</option>
        <option value="B&W">Black & White</option>
      </select>

      <label htmlFor="orientationValue" id="orientationValue">
        Orientation
      </label>
      <select
        type="select"
        id="orientationValue"
        value={orientationValue}
        onChange={event => setOrientationValue(event.target.value)}
      >
        <option value="" disabled selected>
          Select an option
        </option>
        <option value="">None</option>
        <option value="Landscape">Landscape</option>
        <option value="Portrait">Portrait</option>
        <option value="Square">Square</option>
      </select>

      <label htmlFor="peopleValue" id="peopleValue">
        People
      </label>
      <select
        type="select"
        id="peopleValue"
        value={peopleValue}
        onChange={event => setPeopleValue(event.target.value)}
      >
        <option value="" disabled selected>
          Select an option
        </option>
        <option value="">None</option>
        <option value="Yes">Yes</option>
        <option value="No">No</option>
      </select>

      <label id="mediumValue">Medium</label>
      <input
        type="text"
        id="mediumValue"
        value={mediumValue}
        onChange={event => setMediumValue(event.target.value)}
      ></input>
    </form>
  );
};

Filter.propTypes = {
  searchResult: PropTypes.object,
  setSearchResult: PropTypes.func,
  filteredResult: PropTypes.object,
  setFilteredResult: PropTypes.func
};

export default Filter;
