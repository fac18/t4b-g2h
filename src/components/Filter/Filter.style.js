import styled from "styled-components";

const Sticky = styled.div`
  background-color: #4e535f;
  /* z-index: 2000; */
  /* position: fixed;
  top: 8.09rem; */
`;

export { Sticky };
