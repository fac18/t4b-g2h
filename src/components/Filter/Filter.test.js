import React from "react";
import { render } from "@testing-library/react";
import Filter from "./Filter";

it("renders the Filter page", () => {
  const { getByText } = render(<Filter />);
  const titleText = getByText(/collection/i);
  expect(titleText).toBeInTheDocument();
});
