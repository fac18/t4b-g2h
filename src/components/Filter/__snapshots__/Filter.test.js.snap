// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Filter page 1`] = `/collection/i`;
